src/targets/gpu/lowering.cpp

```diff
@@ -361,29 +361,16 @@ struct miopen_apply
     }
 
     /**
-     * Turns on use_local_alloc in the select_module submodules.
-     * Changes the submodule returns to a hip::sync_stream.
+     * Adds dynamic allocation for submodule output parameter.
      */
     void add_select_module_op()
     {
         apply_map.emplace("select_module", [=](instruction_ref ins) {
+            auto s                              = ins->get_shape();
+            auto output                         = insert_allocation(ins, s);
             std::vector<instruction_ref> inputs = ins->inputs();
-            auto mod_args                       = ins->module_inputs();
-            for(auto* smod : mod_args)
-            {
-                smod->use_local_alloc = true;
-                auto last_ins         = std::prev(smod->end());
-                if(last_ins->name() == "@return")
-                {
-                    for(auto out_ins : last_ins->inputs())
-                    {
-                        auto sync_out = smod->insert_instruction(
-                            last_ins, make_op("hip::sync_stream"), out_ins);
-                        smod->replace_return({sync_out});
-                    }
-                }
-            }
-            return ins;
+            inputs.push_back(output);
+            return mod->replace_instruction(ins, ins->get_operator(), inputs, ins->module_inputs());
```

Comment:
Is this section part of the optimums change?

Reply:
Do we have a plan for how the optimums are to be used and, therefore, how a good optimum would be chosen for various ops' outputs?   I see a lot of code for enabling this vector but no idea how it's going to be used.

Reply:
The opts will have to be supplied by the user. They should propagate through the model if possible. The idea being that we will optimize the code to run as fast as possible for the given optimal sizes while other sizes in the range can be slower.